From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Otero <daniel.otero@appentra.com>
Date: Wed, 19 Mar 2025 13:55:30 +0100
Subject: Unalias keywords that are `identifier`s

so they don't show as unnamed children of `identifier`s nodes.
---
 grammar.js                  | 126 ++++++++++++++++++------------------
 test/corpus/expressions.txt |  31 +++++++++
 2 files changed, 94 insertions(+), 63 deletions(-)

diff --git a/grammar.js b/grammar.js
index a89ba01..4e6a7f1 100644
--- a/grammar.js
+++ b/grammar.js
@@ -2281,69 +2281,69 @@ module.exports = grammar({
     // add the keywords here -- and possibly in `conflicts` too.
     identifier: $ => choice(
       /[a-zA-Z_$][\w$]*/,
-      caseInsensitive('allocatable'),
-      caseInsensitive('asynchronous'),
-      caseInsensitive('automatic'),
-      caseInsensitive('block'),
-      caseInsensitive('byte'),
-      prec(-1, caseInsensitive('call')),
-      caseInsensitive('change'),
-      caseInsensitive('constant'),
-      caseInsensitive('contiguous'),
-      caseInsensitive('critical'),
-      caseInsensitive('cycle'),
-      caseInsensitive('data'),
-      caseInsensitive('device'),
-      prec(-1, caseInsensitive('dimension')),
-      caseInsensitive('double'),
-      caseInsensitive('else'),
-      caseInsensitive('elseif'),
-      caseInsensitive('end'),
-      caseInsensitive('endif'),
-      caseInsensitive('entry'),
-      caseInsensitive('error'),
-      caseInsensitive('event'),
-      caseInsensitive('exit'),
-      caseInsensitive('external'),
-      caseInsensitive('fail'),
-      prec(-1, caseInsensitive('flush')),
-      caseInsensitive('form'),
-      caseInsensitive('format'),
-      caseInsensitive('go'),
-      caseInsensitive('if'),
-      prec(-1, caseInsensitive('include')),
-      caseInsensitive('inquire'),
-      caseInsensitive('intrinsic'),
-      caseInsensitive('kind'),
-      caseInsensitive('len'),
-      caseInsensitive('lock'),
-      caseInsensitive('null'),
-      prec(-1, caseInsensitive('open')),
-      caseInsensitive('optional'),
-      caseInsensitive('parameter'),
-      caseInsensitive('pointer'),
-      prec(-1, caseInsensitive('print')),
-      caseInsensitive('private'),
-      caseInsensitive('public'),
-      prec(-1, caseInsensitive('rank')),
-      caseInsensitive('read'),
-      caseInsensitive('real'),
-      caseInsensitive('save'),
-      caseInsensitive('select'),
-      caseInsensitive('sequence'),
-      caseInsensitive('shared'),
-      caseInsensitive('static'),
-      caseInsensitive('stop'),
-      caseInsensitive('sync'),
-      caseInsensitive('target'),
-      caseInsensitive('texture'),
-      prec(-1, caseInsensitive('type')),
-      caseInsensitive('unit'),
-      caseInsensitive('unlock'),
-      caseInsensitive('value'),
-      prec(-1, caseInsensitive('wait')),
-      prec(-1, caseInsensitive('where')),
-      caseInsensitive('write'),
+      caseInsensitive('allocatable', false),
+      caseInsensitive('asynchronous', false),
+      caseInsensitive('automatic', false),
+      caseInsensitive('block', false),
+      caseInsensitive('byte', false),
+      prec(-1, caseInsensitive('call', false)),
+      caseInsensitive('change', false),
+      caseInsensitive('constant', false),
+      caseInsensitive('contiguous', false),
+      caseInsensitive('critical', false),
+      caseInsensitive('cycle', false),
+      caseInsensitive('data', false),
+      caseInsensitive('device', false),
+      prec(-1, caseInsensitive('dimension', false)),
+      caseInsensitive('double', false),
+      caseInsensitive('else', false),
+      caseInsensitive('elseif', false),
+      caseInsensitive('end', false),
+      caseInsensitive('endif', false),
+      caseInsensitive('entry', false),
+      caseInsensitive('error', false),
+      caseInsensitive('event', false),
+      caseInsensitive('exit', false),
+      caseInsensitive('external', false),
+      caseInsensitive('fail', false),
+      prec(-1, caseInsensitive('flush', false)),
+      caseInsensitive('form', false),
+      caseInsensitive('format', false),
+      caseInsensitive('go', false),
+      caseInsensitive('if', false),
+      prec(-1, caseInsensitive('include', false)),
+      caseInsensitive('inquire', false),
+      caseInsensitive('intrinsic', false),
+      caseInsensitive('kind', false),
+      caseInsensitive('len', false),
+      caseInsensitive('lock', false),
+      caseInsensitive('null', false),
+      prec(-1, caseInsensitive('open', false)),
+      caseInsensitive('optional', false),
+      caseInsensitive('parameter', false),
+      caseInsensitive('pointer', false),
+      prec(-1, caseInsensitive('print', false)),
+      caseInsensitive('private', false),
+      caseInsensitive('public', false),
+      prec(-1, caseInsensitive('rank', false)),
+      caseInsensitive('read', false),
+      caseInsensitive('real', false),
+      caseInsensitive('save', false),
+      caseInsensitive('select', false),
+      caseInsensitive('sequence', false),
+      caseInsensitive('shared', false),
+      caseInsensitive('static', false),
+      caseInsensitive('stop', false),
+      caseInsensitive('sync', false),
+      caseInsensitive('target', false),
+      caseInsensitive('texture', false),
+      prec(-1, caseInsensitive('type', false)),
+      caseInsensitive('unit', false),
+      caseInsensitive('unlock', false),
+      caseInsensitive('value', false),
+      prec(-1, caseInsensitive('wait', false)),
+      prec(-1, caseInsensitive('where', false)),
+      caseInsensitive('write', false),
     ),
 
     comment: $ => token(seq('!', /.*/)),
diff --git a/test/corpus/expressions.txt b/test/corpus/expressions.txt
index 7032b51..c5a8f3d 100644
--- a/test/corpus/expressions.txt
+++ b/test/corpus/expressions.txt
@@ -1348,3 +1348,34 @@ end program test
     (comment)
     (end_program_statement
       (name))))
+
+================================================================================
+Reserved keyword as a symbol name
+================================================================================
+
+program main
+  integer :: allocatable(10, 20)
+  allocatable(1, 1) = 0
+end
+
+--------------------------------------------------------------------------------
+
+(translation_unit
+  (program
+    (program_statement
+      (name))
+    (variable_declaration
+      (intrinsic_type)
+      (sized_declarator
+        (identifier)
+        (size
+          (number_literal)
+          (number_literal))))
+    (assignment_statement
+      (call_expression
+        (identifier)
+        (argument_list
+          (number_literal)
+          (number_literal)))
+      (number_literal))
+    (end_program_statement)))
